### PR TITLE
Left view of tree

### PR DESCRIPTION
**Pull request by satyampgt4**
### Have you read the [Contributing Guidelines on Pull Requests](https://github.com/TesseractCoding/NeoAlgo/blob/master/CONTRIBUTING.md#pull-requests)?

yes

### Description

It Will Print Left view of tree

### Checklist

- [x] I've read the contribution guidelines.
- [x] I've checked the issue list before deciding what to submit.
- [x] I've edited the `README.md` and link to my code.

## Related Issues or Pull Requests

#1922

### CODE REVIEW

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@satyampgt4 Rename the program to `left_view_of_tree.cpp`.

**Comment by satyampgt4** [comment]
@iamrajiv  requested changes are done please check

**Comment by satyampgt4** [comment]
@iamrajiv requested changes are done please check

**Review by iamrajiv — CHANGES_REQUESTED** [review]
@satyampgt4 Add space and time complexity.

**Comment by satyampgt4** [comment]
@iamrajiv Time and Space Complexity is added please check

File: C-Plus-Plus/graphs/left_view_of_tree.cpp [hunk 1]
```diff
@@ -0,0 +1,136 @@
+/*
+   This program will print Left view of tree by using concept of Depth First Search with some modification
+   Time Complexity: O(n) 
+   Space Complexity: O(n)
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/left_view_of_tree.cpp]: Push the Time and Space Complexity to the bottom of the Code

File: C-Plus-Plus/graphs/left_view_of_tree.cpp [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+			node *tnode = new node(x);
+			root = tnode;
+		}
+		//if tree if not empty
+		else {
+			node *temp = root;
+			while (1) {
+				//if data to be inserted is less then the data of current node
+				if (x < temp->data ) {
+					//if left child of current node is node present we simple add new node with the data to be inserted
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/left_view_of_tree.cpp]: Break down the Comment into simpler multiple-lines.

File: C-Plus-Plus/graphs/left_view_of_tree.cpp [hunk 3]
```diff
@@ -0,0 +1,136 @@
...
+						break;
+					}
+					//else we change the current node pointer with pointer of left child node pointer
+					else
+						temp = temp->left_child;
+				}
+				//if data to be inserted is grater then or equal to the data of current node
+				else {
+					//if right child of current node is node present we simple add new node with the data to be inserted
+					if (!temp->right_child) {
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/left_view_of_tree.cpp:51]: Break down the Comment into simpler multiple-lines.

File: C-Plus-Plus/graphs/left_view_of_tree.cpp [hunk 4]
```diff
@@ -0,0 +1,136 @@
...
+					if (!temp->left_child) {
+						node *tnode = new node(x);
+						temp->left_child = tnode;
+						break;
+					}
+					//else we change the current node pointer with pointer of left child node pointer
+					else
+						temp = temp->left_child;
+				}
+				//if data to be inserted is grater then or equal to the data of current node
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/left_view_of_tree.cpp]: Break down the Comment into simpler multiple-lines.

File: C-Plus-Plus/graphs/left_view_of_tree.cpp [hunk 5]
```diff
@@ -0,0 +1,136 @@
...
+				}
+			}
+		}
+	}
+	void left_view(node* , int& , int );
+	void left_view_of_tree();
+};
+
+void bin_tree :: left_view_of_tree() {
+	// last printed level
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/left_view_of_tree.cpp]: Remove redundant comments

File: C-Plus-Plus/graphs/left_view_of_tree.cpp [hunk 6]
```diff
@@ -0,0 +1,136 @@
...
+	int last = 0;
+	// current level
```
Comment (HarshCasper) [inline C-Plus-Plus/graphs/left_view_of_tree.cpp]: Remove redundant comments

**Review by HarshCasper — CHANGES_REQUESTED** [review]
Tiny Changes. Would be fit to merge after them 👍

**Comment by satyampgt4** [comment]
@HarshCasper requested changes are done please check

**Review by iamrajiv — APPROVED** [review]

**Review by atarax665 — APPROVED** [review]

**Pull request merged**